### PR TITLE
Components with Environment details page

### PR DESCRIPTION
**Pull request by Sandhya1874**

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import setUpValidationMiddleware from './middleware/setUpValidationMiddleware'
 import indexRoutes from './routes'
 import productRoutes from './routes/products'
 import componentRoutes from './routes/components'
+import componentEnvRoutes from './routes/componentsEnvDetails'
```
Comment (psoleckimoj) [inline server/app.ts:19]: Personally I'd expand `env` or `Env` to environment so it is clear.

Comment (Sandhya1874) [reply server/app.ts:19]: Updated

File: server/app.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import setUpValidationMiddleware from './middleware/setUpValidationMiddleware'
 import indexRoutes from './routes'
 import productRoutes from './routes/products'
 import componentRoutes from './routes/components'
-import componentEnvRoutes from './routes/componentsEnvDetails'
+import componentEnvironmentRoutes from './routes/componentsEnvDetails'
```
Comment (psoleckimoj) [inline server/app.ts:19]: This was for everything where you're using `componentsEnv`, including the filenames etc. Just makes it clear.

Comment (Sandhya1874) [reply server/app.ts]: Updated all file names and references to componentsEnvironment

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: assets/js/componentsEnvironmentDetails.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          healthPath: env.health_path === null ? 'Not set' : env.health_path,
+          appurl: env.url === null ? 'Not set' : env.url,
+          monitor: env.monitor === null ? 'Not set' : env.monitor.toString(),
+          active_agencies: env.active_agencies,
+          ip_allow_list_enabled: env.ip_allow_list_enabled === null ? 'Not set' : env.ip_allow_list_enabled.toString(),
+          include_in_subject_access_requests:
+            env.include_in_subject_access_requests === null
+              ? 'Not set'
+              : env.include_in_subject_access_requests.toString(),
+          modsecurity_enabled: env.modsecurity_enabled === null ? 'Not set' : env.modsecurity_enabled.toString(),
```
Comment (andrewrlee) [inline assets/js/componentsEnvironmentDetails.js:20]: do you need to explicitly call toString on these? I'd have thought they'd be ok

**Pull request merged**